File: PRACTICA DE CAMPO 02-S04/PYTHON/s04_pc02_reto_estadistica_nota.py
total_notas=0
suma_notas=0
nota_alta=0
nota_baja=20
promedio=0
valor_ingresado=0

while True:
    valor_ingresado = float(input("Escribir nota del 0 al 20(-1 para terminar):"))
    if valor_ingresado>=0 and valor_ingresado<=20:
        if valor_ingresado>=17 and valor_ingresado<=20:
            total_notas=total_notas+1
            suma_notas=suma_notas+valor_ingresado
            if valor_ingresado>nota_alta:
                nota_alta=valor_ingresado
            if valor_ingresado<nota_baja:
                nota_baja=valor_ingresado
        elif valor_ingresado>=14 and valor_ingresado<=16:
            total_notas=total_notas+1
            suma_notas=suma_notas+valor_ingresado
            if valor_ingresado>nota_alta:
                nota_alta=valor_ingresado
            if valor_ingresado<nota_baja:
                nota_baja=valor_ingresado
        elif valor_ingresado>=11 and valor_ingresado<=13:
            total_notas=total_notas+1
            suma_notas=suma_notas+valor_ingresado
            if valor_ingresado>nota_alta:
                nota_alta=valor_ingresado
            if valor_ingresado<nota_baja:
                nota_baja=valor_ingresado
        elif valor_ingresado>=0 and valor_ingresado<=10:
            total_notas=total_notas+1
            suma_notas=suma_notas+valor_ingresado
            if valor_ingresado>nota_alta:
                nota_alta=valor_ingresado
            if valor_ingresado<nota_baja:
                nota_baja=valor_ingresado
        else:
            print("Ingrese nota valida.")
    elif valor_ingresado==-1:
        break

promedio = suma_notas/total_notas
print("Total Notas :", total_notas)
print("Suma Notas :", suma_notas)
print("Promedio :", promedio)
print("Nota Alta :", nota_alta)
print("Nota Baja :", nota_baja)
print("Conteo por categoria :",total_notas)

File: PRACTICA DE CAMPO 02-S04/C#/PracticaCampo_02_S04/s04_pc02_reto_estadistica_nota.cs
int total_notas=0;
double suma_notas=0;
double nota_alta=0;
double nota_baja=20;
double promedio=0;
double valor_ingresado;

do{
    Console.WriteLine("Escribir nota del 0 al 20 :");
    valor_ingresado = double.Parse(Console.ReadLine());
    if (valor_ingresado>=0 && valor_ingresado<=20){
        if (valor_ingresado>=17 && valor_ingresado<=20){
            total_notas=total_notas+1;
			suma_notas=suma_notas+valor_ingresado;
            if(valor_ingresado>nota_alta){
                nota_alta=valor_ingresado;
            }
            if(valor_ingresado<nota_baja){
                nota_baja=valor_ingresado;
            }
        }else if (valor_ingresado>=14 && valor_ingresado<=16){
            total_notas=total_notas+1;
			suma_notas=suma_notas+valor_ingresado;
            if(valor_ingresado>nota_alta){
                nota_alta=valor_ingresado;
            }
            if(valor_ingresado<nota_baja){
                nota_baja=valor_ingresado;
            }
        }else if (valor_ingresado>=11 && valor_ingresado<=13){
            total_notas=total_notas+1;
			suma_notas=suma_notas+valor_ingresado;
            if(valor_ingresado>nota_alta){
                nota_alta=valor_ingresado;
            }
            if(valor_ingresado<nota_baja){
                nota_baja=valor_ingresado;
            }
        }else if (valor_ingresado>=0 && valor_ingresado<=10){
            total_notas=total_notas+1;
			suma_notas=suma_notas+valor_ingresado;
            if(valor_ingresado>nota_alta){
                nota_alta=valor_ingresado;
            }
            if(valor_ingresado<nota_baja){
                nota_baja=valor_ingresado;
            }
        }else{
            Console.WriteLine("Ingrese nota valida.");
        }
        promedio = suma_notas/total_notas;
    }
}while (valor_ingresado != -1);

Console.WriteLine("Total Notas : " + total_notas);
Console.WriteLine("Suma Notas : " + suma_notas);
Console.WriteLine("Promedio Notas : " + promedio);
Console.WriteLine("Nota Alta : " + nota_alta);
Console.WriteLine("Nota Baja : " + nota_baja);
Console.WriteLine("Conteo por categoría : " + total_notas);
